fuzz/fuzz_targets/differential.rs

```diff
@@ -37,20 +61,35 @@ fn run(data: &[u8]) -> Result<()> {
     let mut config: Config = u.arbitrary()?;
     config.set_differential_config();
 
-    // Generate the Wasm module.
-    let wasm = if u.arbitrary()? {
+    // Generate the Wasm module; this is specified by either the ALLOWED_MODULES
+    // environment variable or a random selection between wasm-smith and
+    // single-inst.
+    let build_wasm_smith_module = |u: &mut Unstructured, config: &mut Config| -> Result<_> {
         STATS.wasm_smith_modules.fetch_add(1, SeqCst);
-        let module = config.generate(&mut u, Some(1000))?;
-        module.to_bytes()
-    } else {
+        let module = config.generate(u, Some(1000))?;
+        Ok(module.to_bytes())
+    };
+    let build_single_inst_module = |u: &mut Unstructured, config: &mut Config| -> Result<_> {
         STATS.single_instruction_modules.fetch_add(1, SeqCst);
-        let module = SingleInstModule::new(&mut u, &mut config.module_config)?;
-        module.to_bytes()
+        let module = SingleInstModule::new(u, &mut config.module_config)?;
+        Ok(module.to_bytes())
+    };
+    let wasm = match unsafe { &ALLOWED_MODULES.as_slice() } {
+        &[] => panic!("unable to generate a module to fuzz against; check `ALLOWED_MODULES`"),
+        &[n] if n == "wasm-smith" => build_wasm_smith_module(&mut u, &mut config)?,
+        &[n] if n == "single-inst" => build_single_inst_module(&mut u, &mut config)?,
+        _ => {
+            if u.arbitrary()? {
+                build_wasm_smith_module(&mut u, &mut config)?
+            } else {
+                build_single_inst_module(&mut u, &mut config)?
+            }
+        }
     };
```

Comment:
I made the suggested change but added an extra check to `panic!` if `ALLOWED_MODULES.is_empty()`. This can be empty, if the user were to write `ALLOWED_MODULES=-wasm-smith,-single-inst`, so I want to fail with some kind of message in that case.

Reply:
Oh, that's true! Well, you could replace the `.unwrap()` in my suggestion with

```rust
.expect("unable to generate a module to fuzz against; check `ALLOWED_MODULES`")
```

but making the panic explicit is fine too.